src/NFTMarkerCreator.js

```diff
@@ -318,26 +316,22 @@ async function useJPG_w_sharp(buf) {
             imageData.sizeX = metadata.width;
             imageData.sizeY = metadata.height;
             imageData.nc = metadata.channels;
-            return image.toBuffer()
+            return image
+            .raw()
+            .toBuffer()
         })   
         .then(data => {
-            let newArr = [];
+            let dt = data.buffer
 
-            let verifyColorSpace = detectColorSpace(data);
+            let verifyColorSpace = detectColorSpace(dt);
 
             if (verifyColorSpace === 1) {
-                for (let j = 0; j < data.length; j += 4) {
-                    newArr.push(data[j]);
-                }
+                console.log("Color Space is 1 channel!");
             } else if (verifyColorSpace === 3) {
-                for (let j = 0; j < data.length; j += 4) {
-                    newArr.push(data[j]);
-                    newArr.push(data[j + 1]);
-                    newArr.push(data[j + 2]);
-                }
+                console.log("Color Space is 3 channels!");
             }
 
-            let uint = new Uint8Array(newArr);
+            let uint = new Uint8Array(dt);
             imageData.nc = verifyColorSpace;
```

Comment:
i will add a check if number channels from exif is equal to verifyColorSpace the app will pass without an alert, otherwise a message will be dispalyed, with a prompt to continue or not, I will code this with `enquirer` package: